app/models/game.rb

```diff
@@ -28,6 +28,6 @@ def populate_game!
   end
 
   def piece_at(x,y)
-    pieces.find_by(x_coord: x, y_coord: y)
+    pieces.find_by(x_coord: x, y_coord: y) || EmptySquare.new
```

Comment:
@Marsha331 be aware though that this method will not only be used by our view.. And in some cases we rely on the method returning `nil`

Any ideas on how to address it?

Reply:
@Marsha331 After giving it a bit more thought - I'd like to keep the EmptySquare. It is a really nice approach. Though maybe a bit too much for our case 😄 - but remember it for future apps!
So I'd propose the following changes:
adding a method on Game.rb - sth like:
```
def board_piece_at(x,y)
  piece_at(x,y) || EmptySquare.new
end
```

And you call this method from the helper.. This way we separate the logic of a Piece from a 'Board piece' - which serves the only purpose of giving us us the string to display..

What do you think?